components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/ApplicationDAOImpl.java

```diff
@@ -70,7 +75,23 @@ public int addApplication(Application application, int tenantId) throws DeviceMa
         } catch (SQLException e) {
             throw new DeviceManagementDAOException("Error occurred while adding application '" +
                     application.getName() + "'", e);
+        } catch (IOException e) {
+            throw new DeviceManagementDAOException("Error occurred while serializing application properties object", e);
         } finally {
+            if (bao != null) {
+                try {
+                    bao.close();
+                } catch (IOException e) {
+                    log.warn("Error occurred while closing ByteArrayOutputStream", e);
```

Comment:
@dilee shouldn't this be an error log?

Reply:
Fixed in https://github.com/wso2/carbon-device-mgt/pull/950